packages/editor/src/components/post-actions/index.js

```diff
@@ -20,38 +20,62 @@ import { usePostActions } from './actions';
 
 const { Menu, kebabCase } = unlock( componentsPrivateApis );
 
-export default function PostActions( { postType, postId, onActionPerformed } ) {
-	const [ activeModalAction, setActiveModalAction ] = useState( null );
-	const { item, permissions } = useSelect(
+function useEditedEntityRecordsWithPermissions( postType, postIds ) {
```

Comment:
Moved this logic to its own hook. It gets all the edited records and adds permissions.
It's similar to this hook: https://github.com/WordPress/gutenberg/blob/1e28ca49aaedb395cd58abc2c11845ee97c05327/packages/core-data/src/hooks/use-entity-records.ts#L158 but than for `edited` records specifically.

Would it be worth moving to the core-data package as well?